app/models/article_batch.rb

```diff
@@ -0,0 +1,23 @@
+# Copyright 2014 Square Inc.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+
+# Represents the relationship between articles and their batches.
+class ArticleBatch < ActiveRecord::Base
+  belongs_to :article, inverse_of: :article_batches
+
+  # Returns the articles in current batch, with the order of creation (order of ID)
+  def related_articles
+    ArticleBatch.where(batch_name: self.batch_name).order(id: :asc).map(&:article)
```

Comment:
Consider add `includes(:article)` into the query. It will prevent N+1 queries to be performed.